storage/store.go

```diff
@@ -1075,8 +1086,13 @@ func (s *Store) ProcessRangeDescriptorUpdate(rng *Range) error {
 	}
 	delete(s.uninitRanges, rng.Desc().RaftID)
 
-	s.rangesByKey = append(s.rangesByKey, rng)
-	sort.Sort(s.rangesByKey)
+	if s.rangesByKey.Has((*rangeBTreeItem)(rng)) {
+		return &rangeAlreadyExists{rng}
+	}
+	if exRngItem := s.rangesByKey.ReplaceOrInsert((*rangeBTreeItem)(rng)); exRngItem != nil {
```

Comment:
`ReplaceOrInsert` doesn't return an error. From [the docs](https://godoc.org/github.com/google/btree#BTree.ReplaceOrInsert):
> ReplaceOrInsert adds the given item to the tree. If an item in the tree already equals the given one, it is removed from the tree and returned. Otherwise, nil is returned.